internal/provision/devices.go

```diff
@@ -24,24 +29,66 @@ import (
 	"github.com/edgexfoundry/device-sdk-go/v2/internal/container"
 )
 
-func LoadDevices(deviceList []config.DeviceConfig, dic *di.Container) errors.EdgeX {
+func LoadDevices(path string, dic *di.Container) errors.EdgeX {
+	if path == "" {
+		return nil
+	}
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
-	serviceName := container.DeviceServiceFrom(dic.Get).Name
-	var addDevicesReq []requests.AddDeviceRequest
 
-	lc.Debug("loading pre-defined devices from configuration")
-	for _, d := range deviceList {
-		if _, ok := cache.Devices().ForName(d.Name); ok {
-			lc.Debugf("device %s exists, using the existing one", d.Name)
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to create absolute path", err)
+	}
+
+	fileInfo, err := ioutil.ReadDir(absPath)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to read directory", err)
+	}
+
+	var addDevicesReq []requests.AddDeviceRequest
+	lc.Infof("Loading pre-defined devices from %s", absPath)
+	for _, file := range fileInfo {
+		fullPath := filepath.Join(absPath, file.Name())
+		content, err := ioutil.ReadFile(fullPath)
+		if err != nil {
+			lc.Errorf("Failed to read %s: %v", fullPath, err)
 			continue
-		} else {
-			lc.Debugf("device %s doesn't exist, creating a new one", d.Name)
-			deviceDTO, err := createDeviceDTO(serviceName, d, dic)
+		}
+
+		var devices []config.DeviceConfig
```

Comment:
remove the config.DeviceConfig and use dto.device in go-mod-contract, because we might modify it in the future and we should only need to modify it from one place.